tests/functional/mesh/work_test.go

```diff
@@ -536,6 +536,61 @@ func TestWork(t *testing.T) {
 				t.Fatal(err)
 			}
 		})
+		t.Run(testGroup+"/work submit and release to non-existent node", func(t *testing.T) {
+			t.Parallel()
+			if strings.Contains(t.Name(), "kube") {
+				checkSkipKube(t)
+			}
+			controllers, m, _ := workSetup(t.Name())
+			defer tearDown(controllers, m)
+			nodes := m.Nodes()
+
+			// submit work from node1 to non-existent-node
+			// node999 was never initialised
+			unitID, err := controllers["node1"].WorkSubmit("node999", "echosleeplong")
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// wait for 10 seconds
+			ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
+			err = m.WaitForReady(ctx)
```

Comment:
Not sure we need this WaitForReady section -- the mesh is already up and running at this point. We do need the WaitForReady down below though, after the node restart.

Reply:
that said, we probably want to make sure the work is in pending state

```golang
			ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
			err = controllers["node1"].AssertWorkPending(ctx, unitID)
			if err != nil {
				t.Fatal(err)
			}
```

Reply:
I have made the required changes